src/vm/reflectioninvocation.cpp

```diff
@@ -1431,8 +1431,18 @@ FCIMPL4(Object*, RuntimeMethodHandle::InvokeMethod,
         }
 #endif
 
+#ifdef _TARGET_ARM64_
```

Comment:
Are you suggesting to add fields m_hasArgLocDescForStructInRegs & m_argLocDescForStructInRegs in ArgIterator for arm64 as well? These are not required for any other architecture except unix amd64. It is possible on arm64 to get ArgLocDesc with just the offset address. This is how it is for other architectures as well you first get the offset and then get ArgLocDesc. As you said that on amd64 unix that is not possible so I would let the special case be just for amd64 unix and let the logic for arm64 be same as other arch.